test/unit/log.provider.test.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { sinon } from '@loopback/testlab';
+import { Context } from '@loopback/context';
+import { ParsedRequest } from '@loopback/core';
+import { ExtensionStarterBindings, LogProvider } from '../../src';
+
+describe('LogProvider', () => {
+  it('logs a value without a start time', () => {
+    const match = '/test : () => test message';
+
+    runTest(match);
+  });
+
+  it('logs a value with a start time', () => {
+    const match = 'ms: /test : () => test message';
+    const startTime = process.hrtime();
+
+    runTest(match, startTime);
+  });
+});
+
+function runTest(match: string, startTime?: [number, number]) {
+  const ctx: Context = new Context();
+  ctx.bind(ExtensionStarterBindings.LOG_ACTION).toProvider(LogProvider);
+
+  const log = ctx.getSync(ExtensionStarterBindings.LOG_ACTION);
+  const spy = sinon.spy(console, 'log');
+  const req = <ParsedRequest> {url: '/test'};
+
+  if (startTime) log(req, [], 'test message', startTime);
+  else log(req, [], 'test message');
```

Comment:
I would expect the following should work too:

```
log(req, [], 'test message', startTime)
```

It's a common convention that a missing argument and an argument set to `undefined` are treated the same way.